fiftyone/utils/sam2.py

```diff
@@ -23,19 +24,19 @@
 fou.ensure_torch()
 import torch
 
-sam2 = fou.lazy_import("sam2")
-smvid = fou.lazy_import("sam2.sam2_video_predictor")
-smutil = fou.lazy_import("sam2.utils.misc")
-smip = fou.lazy_import("sam2.sam2_image_predictor")
+sam2 = fou.lazy_import("sam2", callback=lambda: fou.ensure_import("sam2"))
```

Comment:
Unfortunately sam2 is not yet available via `pip`, so we can't add it to the requirements list in the models manifest.

However, adding `fou.ensure_import("sam2")` here and calling `dir(sam2)` in the constructor of each model will cause an informative error to be displayed at runtime if SAM2 isn't installed:

```
ImportError: The requested operation requires that 'sam2' is installed on your machine.

If you think this error is inaccurate, you can set `fiftyone.config.requirement_error_level` to 1 (warning) or 2 (ignore).
See https://docs.voxel51.com/user_guide/config.html for details.
```